Carousel: Add admin option to toggle whether or not to display blurred background image.

andrewserong opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->
<!-- Would you like this feature to be tested by Beta testers?
Please add testing instructions to projects/plugins/jetpack/to-test.md in a new commit as part of your PR. -->

This PR is split out from #20377, to explore putting the feature behind an admin toggle. 

#### Changes proposed in this Pull Request:
<!--- Explain what functional changes your PR includes -->
* Add an option to toggle the background color display for the Carousel (the feature introduced in #20377)
* Currently this defaults to off if no option is present — should it instead default to being switched on?

#### Jetpack product discussion
<!-- If you're an Automattician, include a shortlink to the p2 discussion with Jetpack Product here. -->
<!-- Make sure any changes to existing products have been discussed and agreed upon -->
See #20377 for more on the background color feature

#### Screenshots

From options screen (`/wp-admin/options-media.php`):

![image](https://user-images.githubusercontent.com/14988353/127081044-7fec80d1-c873-4746-b258-076c69307847.png)

From Jetpack settings screen (`/wp-admin/admin.php?page=jetpack#/writing`):

![image](https://user-images.githubusercontent.com/14988353/127081153-bc89ebf6-1c83-4cb7-872a-57247b7b8b5c.png)

#### Does this pull request change what data or activity we track or use?
<!--- If so, please add the "[Status] Needs Privacy Updates" label and explain what changes there are. -->
<!--- Check existing Jetpack support documents for a preview of the information we need. -->
No

#### Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

Before testing the below steps, run the following (assuming you're using `wp-env`) to ensure you do not already have a value set for the option — this will help test that the default behaviour is for the feature to be switched off:

```
wp-env run cli wp option delete carousel_display_slide_background
wp-env run cli wp option get carousel_display_slide_background
```

* With the Carousel module enabled go to `/wp-admin/admin.php?page=jetpack#/writing` and you should see an option to toggle the blurred background image
* With the option switched off, the feature introduced in #20377 should not be present
* With the option switched on, the background color / blurred background image should be displayed
* From the Media options screen (`/wp-admin/options-media.php`) repeat the above steps to toggle the feature and ensure it works as expected

andrewserong committed:
Carousel: Add setting toggle for blurred background image

andrewserong committed:
Add changelog entry

andrewserong committed:
Add additional changelog entry

ramonjd commented:
> Currently this defaults to off if no option is present — should it instead default to being switched on?

This is a good question. It's one of things I wonder if anyone will toggle, or understand what it means in the first place?

As you [mention](https://github.com/Automattic/jetpack/pull/20377#issuecomment-886407433), it might be a surprise to some site owners if we switch it on be default.

Could we start with it off by default, and track the toggle maybe? Or at least give it some time in production for folks to provide feedback (if any)?

Just an optional suggestion to tighten the copy:

`Display colorized slide backgrounds`

stacimc reviewed (changes requested):
Thanks for this @andrewserong, this is fantastic! It tests well for me :) I tested on jetpack via jurassic.tube and on wpcom with a bit of manual work to sync changes:
<img width="665" alt="Screen Shot 2021-07-26 at 11 36 59 AM" src="https://user-images.githubusercontent.com/63313398/127041483-6a4c43bf-a086-4f26-85f3-60b7dd8630ce.png">

> Currently this defaults to off if no option is present

I'm not sure I'm understanding correctly -- it looks like this is enabled by default? I would be okay with this defaulting to on. @ramonjd's suggestion to default off for a bit while gathering feedback also sounds good. I think the effect is fairly subtle in dark mode, but with the white background color it may be a bit surprising for existing carousels like you point out. In either case, having the option available alleviates my concerns :)

> Just an optional suggestion to tighten the copy:
> `Display colorized slide backgrounds`

+1 to this ^ so it's clear we've moved away from blurred images, but otherwise I think this is looking really great! I would love to merge this into the background color PR 💯

andrewserong commented:
Thanks for taking a look Ramon and Staci, and for the copy update advice, that reads much better. I'll update today!

andrewserong committed:
Rename option and labels, set default to false

andrewserong committed:
Update changelog entries

andrewserong commented:
Thanks again for the reviews @stacimc and @ramonjd, I've updated the following:

* Renamed the option to `carousel_display_slide_background`
* Updated the copy for the label text to `Display colorized slide backgrounds`
* Updated to ensure the default behaviour is for the feature to be switched off if the option is not set

> Could we start with it off by default, and track the toggle maybe? Or at least give it some time in production for folks to provide feedback (if any)?

I like the idea of introducing the feature with it switched off, and gathering feedback before making the call on whether or not to default to it switched on, but happy to change this if folks feel otherwise. Much like Staci's comment, just having the option here alleviates my concerns, too!

I've updated the screenshots and test instructions in the PR description, but a quick way to ensure your test site does not have the option set (to test the default behaviour for the toggle) is to run the following via `wp-env` before testing:

```
wp-env run cli wp option delete carousel_display_slide_background
```

Let me know if you'd like any other changes!

glendaviesnz reviewed (approved):
Worked as advertised for me.

There is a display issue with the theme dropdown, but not related to this PR:
<img width="277" alt="Screen Shot 2021-07-27 at 4 41 38 PM" src="https://user-images.githubusercontent.com/3629020/127096682-e7c1e5a1-ca75-446c-8abf-fe7303b70045.png">

It is actually slightly worse in master so you seem to have fixed it slightly
<img width="316" alt="Screen Shot 2021-07-27 at 4 42 36 PM" src="https://user-images.githubusercontent.com/3629020/127096712-b3c90bf0-ed50-4955-b6d8-3d29a157afc2.png">

andrewserong commented:
Thanks for reviewing, Glen! Since Staci sounded happy for us to merge this into #20377, I'll do that now, and we can follow up over on that PR if there's anything else that needs looking at.

> There is a display issue with the theme dropdown, but not related to this PR:

I noticed that, too, but couldn't quite work out what's causing it since the CSS for the dropdown doesn't _look_ like it's changed recently. But yes, we can take a look at that separately.
